phase1-coordinator/src/tests.rs

```diff
@@ -1134,6 +1153,158 @@ fn coordinator_drop_contributor_removes_subsequent_contributions() -> anyhow::Re
+        contributor_2.contribute_to(&coordinator).unwrap();
+        verifier_1.verify(&coordinator).unwrap();
+        verifier_1.verify(&coordinator).unwrap();
+    }
+
+    // Add some more participants to proceed to the next round
+    let test_contributor_3 = create_contributor_test_details("3");
+    let test_contributor_4 = create_contributor_test_details("4");
+    let verifier_2 = create_verifier_test_details("2");
+    coordinator
+        .add_to_queue(test_contributor_3.participant.clone(), 10)
+        .unwrap();
+    coordinator
+        .add_to_queue(test_contributor_4.participant.clone(), 10)
+        .unwrap();
+    coordinator.add_to_queue(verifier_2.participant.clone(), 10).unwrap();
+
+    // Update the ceremony to round 2.
+    coordinator.update().unwrap();
+    assert_eq!(2, coordinator.current_round_height().unwrap());
+    assert_eq!(0, coordinator.number_of_queue_contributors());
+    assert_eq!(0, coordinator.number_of_queue_verifiers());
+}
+
+/// Drops a contributor and updates verifier tasks
+///
+/// Make one contribution and verify it, then drop the
+/// contributor. The tasks of a verifier should be updated
+/// properly
+#[test]
+#[serial]
+fn coordinator_drop_contributor_and_update_verifier_tasks() {
+    // Unwraps are used to find out the exact line which produces the error
+    // When the test returns Result with an Err, the line is unknown
+
+    let parameters = Parameters::Custom((
+        ContributionMode::Chunked,
+        ProvingSystem::Groth16,
+        CurveKind::Bls12_377,
+        1, /* power */
+        2, /* batch_size */
+        2, /* chunk_size */
+    ));
+    let replacement_contributor = create_contributor_test_details("replacement-1");
+    let testing = Testing::from(parameters).coordinator_contributors(&[replacement_contributor.participant.clone()]);
+    let environment = initialize_test_environment(&testing.into());
+    let number_of_chunks = environment.number_of_chunks() as usize;
+
+    // Instantiate a coordinator.
+    let coordinator = Coordinator::new(environment, Box::new(Dummy)).unwrap();
+
+    // Initialize the ceremony to round 0.
+    coordinator.initialize().unwrap();
+    assert_eq!(0, coordinator.current_round_height().unwrap());
+
+    // Add a contributor and verifier to the queue.
+    let contributor_1 = create_contributor_test_details("1");
+    let contributor_2 = create_contributor_test_details("2");
+    let verifier_1 = create_verifier_test_details("1");
+    coordinator.add_to_queue(contributor_1.participant.clone(), 10).unwrap();
+    coordinator.add_to_queue(contributor_2.participant.clone(), 9).unwrap();
+    coordinator.add_to_queue(verifier_1.participant.clone(), 10).unwrap();
+
+    // Update the ceremony to round 1.
+    coordinator.update().unwrap();
+
+    contributor_1.contribute_to(&coordinator).unwrap();
+
+    verifier_1.verify(&coordinator).unwrap();
+
+    let locators = coordinator.drop_participant(&contributor_1.participant).unwrap();
```

Comment:
this is the point at which the verifier's tasks should be updated?

Reply:
yes